ARCH-1096 - delete package versions (#1)
im-open/delete-branch-package-versions: README.md

# delete-branch-package-versions

This action will retrieve the list of versions for a GH Package in the provided organization, then delete any package versions with a name that matches the pre-release format generated by [im-open/git-version-lite] (`major.minor.patch-<sanitized-branch-name>.<formated-date>`).  
  - To sanitize the branch name, the characters `a-z, A-Z, 0-9 and -` are kept and any other character is replaced with `-`.  

If the action runs into an issue deleting a specific package version, it will generate a warning that can be viewed in the Summary section of the workflow rather than failing.  Errors retrieving the package versions will still cause the action to fail though.

## Inputs
| Parameter      | Is Required | Description                                                                                                                                                        |
| -------------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | `github-token` | true        | A token with permission with the `packages:delete` and `packages:read` scopes that is authorized in the organization where the package versions to be deleted are. |

Reviewer: Something about how this reads is throwing me off a bit, but maybe it's nothing. Here's a suggestion of what feels a bit better to me, but I'm not married to the idea of changing this.
```suggestion
| `github-token` | true        | An access token with the `packages:delete` and `packages:read` scopes that is authorized in the organization where the package versions to be deleted are. |
```

> | `organization` | false       | The branch the packages were created for.  Defaults to `github.context.repo.org` if not provided.                                                                  |

Reviewer: The branch the packages were created for? I'm not sure what the wording should be here, but this seems to be a duplicate of the `branch-name` parameter.

Author: It's not accurate, it should be around the organization.  I'll update it.
